### ISSUE TITLE
google_pubsub_subscription.push_subscriptions block should not be added if there are no push subscriptions

### ISSUE THREAD

**Issue by ferrarimarco** [terraform-google-modules/terraform-google-pubsub#81]
### TL;DR

The `google_pubsub_subscription.push_subscriptions` [is empty if you don't specify any push_subscription](https://github.com/terraform-google-modules/terraform-google-pubsub/blob/3743450ff03a238e9fcd840f29bffd178d54d4b4/main.tf#L118), but Terraform reports a diff if you don't specify any.

### Expected behavior

The `push_config` block should not be added to the `google_pubsub_subscription.push_subscriptions` if no push_subscriptions are there because it's an optional field. Maybe making it [dynamic](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html) may solve this?

### Observed behavior

```hcl
  # module.whatever.google_pubsub_subscription.push_subscriptions["whatever"] will be updated in-place
  ~ resource "google_pubsub_subscription" "push_subscriptions" {
        id                         = "projects/AAAAAAAAA/subscriptions/BBBBBBBB-subscription"
        name                       = "whatever"
        # (8 unchanged attributes hidden)


      + push_config {
        }
        # (1 unchanged block hidden)
    }
```

### Terraform Configuration

```hcl
module "whatever_pubsub" {
  project_id = var.google_project_id
  source     = "terraform-google-modules/pubsub/google"
  topic      = "camera-integration"
  version    = "3.0.0"

  push_subscriptions = [
    {
      name          = "whatever"
      push_endpoint = ""
    }
  ]
}
```

### Terraform Version

```sh
Terraform v1.0.9
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v3.84.0
+ provider registry.terraform.io/hashicorp/google-beta v3.88.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
+ provider registry.terraform.io/hashicorp/random v2.3.1
```
```


### Additional information

_No response_

**Comment by ferrarimarco** [comment]
still relevant :)

**Comment by ferrarimarco** [comment]
Not stale
